### ISSUE TITLE
isEmail does not allow +tag addresses

### ISSUE THREAD

**Issue by phlgr** [mantinedev/mantine#3838]
### What package has an issue

@mantine/form

### Describe the bug

The email `test+test@test.com` does not appear to be valid, even though it is a valid email.
`+tag` addresses are commonly used as alternative to aliases and should therefore be recognized as valid email.
The `+` character is also a valid part of the local-part of an email.

### What version of @mantine/hooks page do you have in package.json?

5.10.3

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

### Possible fix

The regex could be extended to recognize a `+` as a valid character.
```diff
- /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/
+ /^\w+([.+-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/
```

**Comment by jeysal** [comment]
I'll just leave this here https://stackoverflow.com/questions/201323/how-can-i-validate-an-email-address-using-a-regular-expression

There are likely a lot of other valid email addresses that are rejected because the current regex is extremely restrictive.

The StackOverflow thread contains suggestions for more correct regexes, but note that they generally don't validate that the domain has multiple parts (technically it's possible to have an email server on a TLD).
Many in the StackOverflow thread also advocate that validating email beyond, say, `includes('@')` is harmful / not worth the effort to do correctly and it should only be done by actually sending an email.

**Comment by rtivital** [comment]
Already fixed, will be available in 6.0.3 release

**Comment by rtivital** [comment]
Fixed in 6.0.4
